src/challenges.ts

```diff
@@ -300,3 +292,45 @@ export function getDifficultyName(difficulty: ChallengeDifficulty): string {
       return 'Unknown';
   }
 }
+
+/**
+ * Assigns the user a role based on their difficulty level.
+ * @param guild - The Discord guild where the roles are located.
+ * @param discordId - The user's Discord ID.
+ * @param difficulty - The difficulty enum for the user.
+ */
+export async function assignSageRole(
+  guild: Guild,
+  discordId: string,
+  difficulty: ChallengeDifficulty,
+): Promise<void> {
+  const member: GuildMember | null = await guild.members.fetch(discordId);
+
+  if (!member) {
+    console.error(
+      `User with ID ${discordId} not found. Could not assign Sage's role`,
+    );
+    return;
+  }
+
+  // Determine role based on difficulty
+  let roleName: string | null = null;
+
+  if (difficulty === ChallengeDifficulty.MASTER) {
+    roleName = "Sage's Master";
+  } else if (difficulty === ChallengeDifficulty.GRANDMASTER) {
+    roleName = "Sage's Grandmaster";
```

Comment:
We should probably make these constants or an enum somewhere so that they are consistent everywhere.